packages/dd-trace/src/appsec/rule_manager.js

```diff
@@ -26,137 +22,66 @@ function loadRules (config) {
 }
 
 function updateWafFromRC ({ toUnapply, toApply, toModify }) {
-  const batch = new Set()
-
-  const newRulesData = new SpyMap(appliedRulesData)
-  let newRuleset
-  let newRulesetId
-  const newRulesOverride = new SpyMap(appliedRulesOverride)
-  const newExclusions = new SpyMap(appliedExclusions)
-  const newCustomRules = new SpyMap(appliedCustomRules)
   const newActions = new SpyMap(appliedActions)
 
+  let wafUpdated = false
+  let wafUpdatedSuccess = true
+
   for (const item of toUnapply) {
-    const { product, id } = item
+    if (!ASM_PRODUCTS.has(item.product)) continue
+
+    try {
+      waf.wafManager.remove(item.path)
+      item.apply_state = ACKNOWLEDGED
+      wafUpdated = true
 
-    if (product === 'ASM_DATA') {
-      newRulesData.delete(id)
-    } else if (product === 'ASM_DD') {
-      if (appliedRulesetId === id) {
-        newRuleset = defaultRules
+      if (item.product === 'ASM') {
+        newActions.delete(item.id)
       }
-    } else if (product === 'ASM') {
-      newRulesOverride.delete(id)
-      newExclusions.delete(id)
-      newCustomRules.delete(id)
-      newActions.delete(id)
+    } catch (e) {
+      wafUpdatedSuccess = false
+      item.apply_state = ERROR
+      item.apply_error = e.toString()
+      Reporter.reportWafConfigError(waf.wafManager.ddwafVersion, waf.wafManager.rulesVersion)
     }
   }
 
   for (const item of [...toApply, ...toModify]) {
```

Comment:
what happens with the new waf if you try to add the same config path twice ?
this would happen when a config item is in toModify. The path would be the same, but the payload changed. Does the WAF accept that or would it require us to delete+readd it ?

Reply:
The new WAF is able to handle this kind of situations - same config path, different payload.